plugins/destination/mysql/client/migrate.go

```diff
@@ -22,6 +22,8 @@ func (c *Client) normalizeTables(tables schema.Tables) schema.Tables {
 func (c *Client) normalizeTable(table *schema.Table) *schema.Table {
 	columns := make([]schema.Column, len(table.Columns))
 	for i, col := range table.Columns {
+		// In MySQL primary keys are implicitly not null
+		col.NotNull = col.NotNull || col.PrimaryKey
```

Comment:
I'm a bit worried that we modify column here instead of working on the copy.
Could we move it to
```go
normalizedCol := schema.NewColumnFromArrowField(*normalized)
normalizedCol.NotNull = normalizedCol.NotNull || normalizedCol.PrimaryKey
columns[i] = normalizedCol
```
or something similar?

Reply:
Cleaned up the PR. Also thought about removing the conversion to arrow and back to column, but probably safer to keep it so we remove any non relevant column fields.

Also moved the tests change to https://github.com/cloudquery/cloudquery/pull/13565